### PR TITLE
Alter bootstrap to handle new FreeBSD pkg changes

### PR DESCRIPTION
**Pull request by lifeisstillgood**
FreeBSD's `pkg` system is under development and has changed
it seems since this bootstrap was written.  A new install
of 9.2-Release will not work with the current version.

There are two notable changes
- use the pkg.freebsd repository, which has support for more
  architectures (ll 2337).  the pcbsd site only supports
  amd64, and I at least still have some i386's.
- use the new format for repository conf files (ll 2362-2366)
  The format of conf files has changed and now needs a more
  YAML/JSON like definition.

This is tested on a clean install of 9.2-RELEASE and as the
very first file downloaded it then runs cleanly and
installs salt.

I hope contribution this helps your project

### CODE REVIEW

File: bootstrap-salt.sh [hunk 1]
```diff
@@ -1735,7 +1733,7 @@ install_fedora_stable() {
 
 install_fedora_git_deps() {
     install_fedora_deps || return 1
-    yum install -y git yum-utils || return 1
+    yum install -y git || return 1
```
Comment (s0undt3ch) [inline bootstrap-salt.sh]: Did you remove `yum-utils` on purpose?

File: bootstrap-salt.sh [hunk 2]
```diff
@@ -1276,9 +1277,6 @@ install_ubuntu_deps() {
 
     apt-get update
 
-    # Minimal systems might not have upstart installed, install it
-    __apt_get_install_noinput upstart
```
Comment (s0undt3ch) [inline bootstrap-salt.sh:1286]: Was this removed on purpose?

File: bootstrap-salt.sh (hunk 1 above)
Comment (lifeisstillgood) [inline bootstrap-salt.sh]: No, I am afraid I missed that one.  I have no idea why I would remove that - I suspect I was using out of date master / someone got in before me.  Its on my pull however so I will update that branch - but not today for obvious reasons.

File: bootstrap-salt.sh (hunk 2 above)
Comment (lifeisstillgood) [inline bootstrap-salt.sh:1286]: Ok - I am now confused.

git diff dfa0f963486465b320330f473622c453c641dc5e  shows the previous commit on branch develop already had those missing.  I have an awful suspicion I have branched off develop and not master as I should have - please point me at the contrib details and I will be happy to review - but after holidays.

Thank you

Comment (s0undt3ch) [inline bootstrap-salt.sh:1286]: Oh, please, always branch off develop, I'll merge to stable when ready.

**Comment by lifeisstillgood** [comment]
I will chalk up my last poorly thought out comment to an excess of Xmas wine:-)
I will rebase against develop and resubmit, but probably not till Friday at earliest

Thank you

**Comment by s0undt3ch** [comment]
Cool!

**Comment by lifeisstillgood** [comment]
@s0undt3ch : I have rebased this commit against HEAD of develop (9387bf48) so it should have few surprises this time 

There is more LOC changes, but conceptually still the same - fetch packages from the "official" repo (2345-2350)
and rewrite the conf file that `pkg` depends on, into the new JSON (ish) (2355-2375) - this is now a seperate function to keep things sane.  I hope the comments make sense

HTH

**Comment by s0undt3ch** [comment]
This can't be merged from GitHub, I'll hand merge it...

**Comment by s0undt3ch** [comment]
Thank You @lifeisstillgood !

**Comment by lifeisstillgood** [comment]
my apologies.
I certainly do not wish to cause you inconvenience again - could you let me
know the cause and I will avoid for next time !
- I assume it was either my repo on GitHub is not in sync with saltstack
  (cannot find a button to do that) or I totally screwed up and still was
  rebased onto out of date revision?

thank you for accepting the patch anyway - I hope to do it smoother next
time :-)

Paul Brian
07540 456 115
paul@mikadosoftware.com
@lifeisstillgood
pbrian.mikadosoftware.com

On 9 Jan 2014, at 19:21, Pedro Algarvio notifications@github.com wrote:

This can't be merged from GitHub, I'll hand merge it...

—
Reply to this email directly or view it on
GitHubhttps://github.com/saltstack/salt-bootstrap/pull/275#issuecomment-31966329
.

## 

www.mikadosoftware.com

See your software improve with every change.

**Comment by s0undt3ch** [comment]
Nah.... No reason for apologies.

GitHub just can't merge a commit, either because of the reasons you evoked, or some others...
I had no conflicts while hand merging it so I can't say what was GitHub's issue.

About still accepting your PR, of course!!! We want this project to grow in the right direction.

**Pull request merged**